#!/bin/sh -e
##
## 03_broadcaster_fixes 
##

. debian/patches/patch-opts

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -r 39cc49096fed -r c0b5b9576c51 src/xine-engine/broadcaster.c
--- a/src/xine-engine/broadcaster.c	Sat Jul 14 14:50:02 2007 +0200
+++ b/src/xine-engine/broadcaster.c	Fri Jul 27 22:24:01 2007 +0300
@@ -352,6 +352,17 @@ broadcaster_t *_x_init_broadcaster(xine_
 
 void _x_close_broadcaster(broadcaster_t *this)
 {
+  this->running = 0;
+  pthread_cancel(this->manager_thread);
+  pthread_join(this->manager_thread,NULL);
+  close(this->msock);
+  
+  if (this->stream->video_fifo)
+    this->stream->video_fifo->unregister_put_cb(this->stream->video_fifo, video_put_cb);
+
+  if(this->stream->audio_fifo)
+    this->stream->audio_fifo->unregister_put_cb(this->stream->audio_fifo, audio_put_cb);
+  
   xine_list_iterator_t ite;
   
   while ( (ite = xine_list_front(this->connections)) ) {
@@ -362,18 +373,9 @@ void _x_close_broadcaster(broadcaster_t 
     xine_list_remove (this->connections, ite);
   }
   xine_list_delete(this->connections);
-  
-  this->running = 0;  
-  close(this->msock);
-  pthread_mutex_lock( &this->lock );
-  pthread_cancel(this->manager_thread);
-  pthread_join(this->manager_thread,NULL);
-    
-  this->stream->video_fifo->unregister_put_cb(this->stream->video_fifo, video_put_cb);
-
-  if(this->stream->audio_fifo)
-    this->stream->audio_fifo->unregister_put_cb(this->stream->audio_fifo, audio_put_cb);
-  
+
+  pthread_mutex_destroy( &this->lock );
+
   free(this);
 }
 
